akka-typed/src/main/scala/akka/typed/Impl.scala

```diff
@@ -20,24 +21,35 @@ private[typed] class ActorAdapter[T](_initialBehavior: () ⇒ Behavior[T]) exten
   val ctx = new ActorContextAdapter[T](context)
 
   def receive = LoggingReceive {
-    case akka.actor.Terminated(ref) ⇒ next(behavior.management(ctx, Terminated(ActorRef(ref))))
-    case akka.actor.ReceiveTimeout  ⇒ next(behavior.management(ctx, ReceiveTimeout))
-    case msg                        ⇒ next(behavior.message(ctx, msg.asInstanceOf[T]))
+    case akka.actor.Terminated(ref) ⇒
+      val msg = Terminated(ActorRef(ref))
+      next(behavior.management(ctx, msg), msg)
+    case akka.actor.ReceiveTimeout ⇒
+      next(behavior.management(ctx, ReceiveTimeout), ReceiveTimeout)
+    case msg ⇒
+      val m = msg.asInstanceOf[T]
+      next(behavior.message(ctx, m), m)
   }
 
-  private def next(b: Behavior[T]): Unit = {
+  private def next(b: Behavior[T], msg: Any): Unit = {
+    if (isUnhandled(b)) unhandled(msg)
     behavior = canonicalize(ctx, b, behavior)
     if (!isAlive(behavior)) {
       context.stop(self)
     }
   }
 
+  override def unhandled(msg: Any): Unit = msg match {
```

Comment:
is this method overridable by users? If so, isn't it dangerous to let them override the DeathPact behavior?